scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/MissingAssociationConflict.java

```
@@ -0,0 +1,28 @@
+package gov.nasa.jpl.aerie.scheduler;
+
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+
+import java.util.Collection;
+
+public class MissingAssociationConflict extends Conflict {
+  private final Collection<ActivityInstance> instances;
+
+  /**
+   * ctor creates a new conflict
+   *
+   * @param goal IN STORED the dissatisfied goal that issued the conflict
+   */
+  public MissingAssociationConflict(final Goal goal, final Collection<ActivityInstance> instancesToChooseFrom) {
+    super(goal);
+    this.instances = instancesToChooseFrom;
+  }
+
+  public Collection<ActivityInstance> getActivityInstancesToChooseFrom(){
+    return instances;
+  }
+
+  @Override
+  public Windows getTemporalContext() {
+    return null;
```

Why is this null? If it's only unclear to me because I'm unfamiliar with this area of our system, a simple answer here will suffice, but if it's less obvious then a code comment may be helpful too.

---

It is because the temporal context here is irrelevant. We could probably throw an exception instead because you should not use the temporal context of a `MissingAssociationConflict`.

---

Just as a note, it's usually good to [avoid using `null`](http://www.javapractices.com/topic/TopicAction.do?Id=134) in situations where it can leak out beyond its context. 

If you shouldn't be calling the method (and you have a way to know this from the outside), an exception is good. Otherwise, an [informationless instance](http://www.javapractices.com/topic/TopicAction.do?Id=59) of Windows is better, e.g. `Windows.forever()` or `new Windows()` (empty).